packages/components/src/components/navigation/model.ts

```diff
@@ -79,15 +79,20 @@ export class MenuItemModel extends Record ({
                 const parts = rawData.key.split("?");
                 const subParts = parts[0]
                     .split("/")
-                    .filter(val => val !== '');
+                    .filter(val => val !== '')
+                    .map(val => QueryKey.decodePart(val));
+
+                const decodedPart = subParts.join('/');
```

Comment:
Comments might be nice here to illustrate why the splitting and joining is necessary.